staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -414,10 +416,20 @@ func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.L
 		return nil, nil
 	}
 	isInternal := requiresInternalLoadBalancer(service)
-	lbFrontendIPConfigName := az.getFrontendIPConfigName(service)
 	serviceName := getServiceName(service)
 	for _, ipConfiguration := range *lb.FrontendIPConfigurations {
-		if lbFrontendIPConfigName == *ipConfiguration.Name {
+		owns, isPrimaryService, err := az.serviceOwnsFrontendIP(ipConfiguration, service)
+		if err != nil {
+			klog.Errorf("get(%s): lb(%s) - failed to filter frontend IP configs with error: %v", serviceName, *lb.Name, err)
+			return nil, nil
+		}
+		if owns {
+			if isPrimaryService {
```

Comment:
could reduce code like this, this is only for logging, not necessary to use if-else
```
klog.Infof("get(%s): lb(%s) - found frontend IP config,  primary service: %v", serviceName, *lb.Name, isPrimaryService)
```

And also specify log level, level 2 should be printed, > 2 is for troubleshooting.